src/host/VtIo.cpp

```diff
@@ -790,3 +791,51 @@ void VtIo::Writer::WriteInfos(til::point target, std::span<const CHAR_INFO> info
         } while (--repeat);
     }
 }
+
+void VtIo::Writer::WriteScreenInfo(SCREEN_INFORMATION& newContext, til::size oldSize) const
+{
+    const auto area = static_cast<size_t>(oldSize.width * oldSize.height);
+
+    auto& main = newContext.GetMainBuffer();
+    auto& alt = newContext.GetActiveBuffer();
+    const auto hasAltBuffer = &alt != &main;
+
+    // TODO GH#5094: This could use xterm's XTWINOPS "\e[8;<height>;<width>t" escape sequence here.
+    if (oldSize != main.GetBufferSize().Dimensions())
+    {
+        THROW_IF_NTSTATUS_FAILED(main.ResizeTraditional(oldSize));
+        main.SetViewportSize(&oldSize);
+    }
+    if (hasAltBuffer && oldSize != alt.GetBufferSize().Dimensions())
+    {
+        THROW_IF_NTSTATUS_FAILED(alt.ResizeTraditional(oldSize));
+        alt.SetViewportSize(&oldSize);
+    }

```

Comment:
These two blocks have changed slightly, to avoid reflow when the size hasn't changed (the common case I assume). All other code remains unchanged.